ocaml/typing/mode.ml

```diff
@@ -49,8 +49,24 @@ module Global_flag = struct
     | Global, Global | Unrestricted, Unrestricted -> 0
 end
 
+module type BiHeyting = sig
+  (** Extend the [Lattice] interface with operations of bi-Heyting algebras *)
+
+  include Lattice
+
+  (** [imply c] is the right adjoint of [meet c]; That is, for any [a] and [b],
+      [meet c a <= b] iff [a <= imply c b] *)
+  val imply : t -> t -> t
```

Comment:
Hmm I guess those would be good names for totally ordered bi-Heyting algebras, but in general they don't necessarily behave that way.

How about this: we stick with these names in `module type BiHeyting` (because it defines bi-heyting algebras in general), but in `module Total`, which is the special case of totally ordered bi-heyting algebras, we create aliases `low_pass` and `high_pass` pointing to `imply` and `subtract`?

Reply:
It's probably not worth fussing about, in the end. These names are used only locally. Let's just merge as is. I'll Resolve.